### ISSUE TITLE
bundler --install fails with permission denied

### ISSUE THREAD

**Issue by pu239ppy** [bloomberg/chef-bach#874]
while running cookbooks/bach_repository/recipes/gems.rb https://github.com/bloomberg/chef-bach/blob/master/cookbooks/bach_repository/recipes/gems.rb#L36 will fail with permission denied.  This is due to /home/vagrant/.bundle being owned by root and not having sufficient privileges to modify the cache.  Temporary workaround is to run this prior

````
diff --git a/cookbooks/bach_repository/recipes/gems.rb b/cookbooks/bach_repository/recipes/gems.rb
index fbce0a5..af15b62 100644
--- a/cookbooks/bach_repository/recipes/gems.rb
+++ b/cookbooks/bach_repository/recipes/gems.rb
@@ -57,6 +57,14 @@ ruby_block 'determine-bundler-command' do
   end
 end

+execute 'permissions hack' do
+  cwd '/home/vagrant'
+  command 'chown -R vagrant:vagrant .bundle || /bin/true '
+end
+
+
+
+
 execute 'bundler install' do
   cwd node['bach']['repository']['repo_directory']
   command lazy { node.run_state[:bcpc_bootstrap_bundler_command] }````

**Comment by cbaenziger** [comment]
It looks like despite both bundler commands are being run as `vagrant` in [gem.rb](https://github.com/bloomberg/chef-bach/blob/c45e32bab089ced1dfb5778d2c36afe7914023f7/cookbooks/bach_repository/recipes/gems.rb#L36-L60) the first bundler command is using `sudo` internally. We are ending up with the following from `auditd` logs:
```
[...]
type=SYSCALL msg=audit(1496899595.169:390): arch=c000003e syscall=87 success=yes exit=0 a0=7f1c041c3f40 a1=7f1c041c6610 a2=0 a3=7f1c0ab6b130 items=2 ppid=26831 pid=27111 auid=900 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=14 comm="bundle" exe="/opt/chef/embedded/bin/ruby" key="foo"
type=SYSCALL msg=audit(1496899595.169:391): arch=c000003e syscall=84 success=yes exit=0 a0=7f1c04003060 a1=7f1c040030db a2=1 a3=3 items=2 ppid=26831 pid=27111 auid=900 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=14 comm="bundle" exe="/opt/chef/embedded/bin/ruby" key="foo"
type=SYSCALL msg=audit(1496899611.501:397): arch=c000003e syscall=83 success=no exit=-17 a0=2788380 a1=1ff a2=902005 a3=3 items=1 ppid=26831 pid=27399 auid=900 uid=900 gid=0 euid=900 suid=900 fsuid=900 egid=0 sgid=0 fsgid=0 tty=(none) ses=14 comm="bundle" exe="/opt/chef/embedded/bin/ruby" key="foo"
type=SYSCALL msg=audit(1496899611.501:398): arch=c000003e syscall=83 success=no exit=-13 a0=2e7abb0 a1=1c0 a2=902005 a3=3 items=2 ppid=26831 pid=27399 auid=900 uid=900 gid=0 euid=900 suid=900 fsuid=900 egid=0 sgid=0 fsgid=0 tty=(none) ses=14 comm="bundle" exe="/opt/chef/embedded/bin/ruby" key="foo"
[...]```

**Comment by vt0r** [comment]
Fixed in #882 with a workaround
